### ISSUE TITLE
Broken images in pages.

### ISSUE THREAD

**Issue by Cramin** [Cacti/plugin_mactrack#52]
I have problem with icon images in mactrack pages.
`<a href='//plugins/mactrack/mactrack_devices.php?action=edit&amp;device_id=1' title='Edit Device'>`
This is a link. I have installed Cacti in root dir in web server.
`/* set the $url_path to point to the default URL of your cacti`
` * install ex: if your cacti install as at http://serverip/cacti/ this would be set to /cacti/.`
`*/`

`$url_path = '/';`

**Comment by jpobeda** [comment]
Uhmm I'm not sure I'm following you. Can you post an screenshot?

**Comment by Cramin** [comment]
https://i.imgur.com/GNrOW89.png
`        $webroot = $config['url_path'] . '/plugins/mactrack/';`
If i remove $config['url_path'] and leave in mactrack_view files
`        $webroot = '/plugins/mactrack/';`
it works.

**Comment by jpobeda** [comment]
Oh you're right but I don't see them broken. I'm using Chrome and Firefox on windows and they both show the icons right.

That's easy to fix in the meantime your workaround it's just fine :P

**Comment by jpobeda** [comment]
Actually I didn't read it properly.

webroot for your cacti installation is / and NOT /cacti/. 

Mactrack reads $config['url_path'] from cacti to get the right webroot.

Maybe @netniV or @cigamit  can look at this.

**Comment by netniV** [comment]
The images that look wrong, can you get the full URL that they are using?  On some browser, you can even right click and image and say Open Image in new tab to get it.

The plugins should always be under the cacti root folder.  So, if your cacti location is not in a subdirectory of a host (ie, it's not http://host/cacti/ but just the host http://host) then the config variable should be set to "/" as above.  

@jpobeda's fix will help because the folder path should always have a trailing / though it shouldn't cause a problem.  The other thing to do is make sure that you have the right ownership and user/group/world permissions on the mactrack plugin subdirectories.

**Comment by Cramin** [comment]
Look at the link in my first post, this is link from source code. Every link to image begins with double /.
Full URL of the broken image is `http://plugins/mactrack/mactrack_devices.php?action=edit&device_id=6`
so where is my IP or domain name?
When delete $config variable i get good links: `http://cacti.domain/plugins/mactrack/mactrack_devices.php?action=edit&device_id=6`

All dir where cacti is have www-data permisions. I use nginx web server.

**Comment by netniV** [comment]
OK.  Couple more questions then.

- Does this happen on every page or just some of them?
- Did you place the mactrack folder directly under `<cacti_folder>/plugins`?  

   The top of every plugin page changes directory up two levels (going from `<cacti_folder>/plugins/mactrack` to `<cacti_folder>/`) and then includes `include/auth.php` which in turns includes all necessary base files required.  

   Because of this you can not use a symbolic link to map to the folder in another location.
- Are there any errors appearing in the php/web/cacti logs?

Basically, $config should be being populated as an array via the auth.php include.  Maybe try adding a bit of code that dumps out the config variable and see if its actually primed.

**Comment by Cramin** [comment]
Hello again.
I make new server. Debian 9.3, MariaDB + PHP7 + apache2 from repo. Cacti + cacti-spine from GIT 1.1.33.
The answers:

- This is happen on all pages, i think that mactrack have problem with cacti in root dir. I have problems in Device Tracking submenu. When editing devices, devices types after submit i have loaded cacti top bar twice. https://ibb.co/nteiKG

- Yes /var/www/html/plugins/mactrack, /var/www/html/ is the root folder where cacti index.php is.
`root@dell1950:/var/www/html/plugins/mactrack# ls -l ../../include/auth.php`
`-rw-r--r-- 1 www-data www-data 8327 sty 23 16:11 ../../include/auth.php`
There are no symbolic links for www root folder, cacti in other place linked to /var/www/html folder.

- Cacti, apache, php logs are clear, no errors related to this.

**Comment by netniV** [comment]
OK.  Double checked this by installing (but not enabling) the plugin which gives me access to the various pages.  Using both the Modern and Paw themes (just to be sure it's not a theme issue) I can't get the page to display what you have. I can get a blank page after clicking create or save (which is again wrong).

Can you give me the exact steps you went through after logging in, so I can try to recreate this?

**Comment by Cramin** [comment]
Only uploaded dir, chowned to www-data:www-data, go to Plugins Manager, Installed and Enabled Mactrack.
I have this on old server Cacti + Nginx, and on two new servers Cacti + apache2.
Exactly the same problem. All instalations are in root dir, not in cacti/ subdir.

**Comment by netniV** [comment]
Have you updated the config.php file to have the correct location?  By default the file has '/cacti/' in it if I remember rightly.

**Comment by cigamit** [comment]
Should be resolved now.  Thanks for reporting.

**Comment by cigamit** [comment]
Sorry commit is here: https://github.com/Cacti/plugin_mactrack/commit/e2e32b86f8fd8651b30af6bdffdd8c69297e429a
